### PR TITLE
Notice fixes in paymentBlock.tpl

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Notice fixes in paymentBlock.tpl

Before
----------------------------------------
image

After
----------------------------------------
image

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: templates/CRM/common/paymentBlock.tpl [hunk 1]
```diff
@@ -85,23 +85,21 @@
       {else}
         {capture assign='preProfileID'}{/capture}
       {/if}
+      {if !array_key_exists('urlPathVar', $form)}{assign var='urlPathVar' value=null}{/if}
```
Comment (larssandergreen) [inline templates/CRM/common/paymentBlock.tpl:88]: Seems a bit confusing to do this one differently than all the others, why not make this the same as billing_profile_id below?

Comment (demeritcowboy) [reply templates/CRM/common/paymentBlock.tpl:88]: Are either of them elements in the $form array? They both seem like top-level smarty vars. I.e. array_key_exists will always be false. I don't see them available in $form with `{debug}`?

(Note there is also a javascript `$form` to confuse things)

Comment (eileenmcnaughton) [reply templates/CRM/common/paymentBlock.tpl:88]: Coming back to this it looks like urlPathVar is almost never set `CRM_Event_Form_Task_Register` seems like the only possibly relevant place

image

Comment (eileenmcnaughton) [reply templates/CRM/common/paymentBlock.tpl:88]: OK - but that is in the chunk of code that calls civicrm/payment/form - ie credit card stuff - which isn't called by The register task based on my debugging

So, from what I can tell that whole `$urlPathVar` does nothing - on the bright side that means that it always being false like @demeritcowboy says would be no change...

File: templates/CRM/common/paymentBlock.tpl [hunk 2]
```diff
@@ -85,23 +85,21 @@
...
         {capture assign='urlPathVar'}{$urlPathVar}&{/capture}
-      {else}
-        {capture assign='urlPathVar'}{/capture}
       {/if}
-      {if $billing_profile_id}
+      {if array_key_exists('billing_profile_id', $form)}
         {capture assign='profilePathVar'}billing_profile_id={$billing_profile_id}&{/capture}
       {else}
-        {capture assign='profilePathVar'}{/capture}
+        {assign var='profilePathVar' value=null}
```
Comment (eileenmcnaughton) [inline templates/CRM/common/paymentBlock.tpl:95]: profilePath var not looking more encouraging 

image

Comment (eileenmcnaughton) [reply templates/CRM/common/paymentBlock.tpl:95]: oh that depends on `billing_profile_id`

Comment (eileenmcnaughton) [reply templates/CRM/common/paymentBlock.tpl:95]: OK - I think this is a better fix for billing_profile_id / profile_pathVar 

https://github.com/civicrm/civicrm-core/pull/26960

**Pull request closed without merging**